### PR TITLE
Ccs 2790 ability to delete modules

### PR DESCRIPTION
**Pull request by xdavidson**

### CODE REVIEW

File: src/main/java/com/redhat/pantheon/servlet/ModuleRootQueryServlet.java [hunk 1]
```diff
@@ -40,14 +42,15 @@
...
+	private final Logger logger = LoggerFactory.getLogger(ModuleRootQueryServlet.class);
+	
     @Override
     protected void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response) throws ServletException, IOException {
 
         // Get the query parameter(s)
         String moduleName = request.getParameter("name");
 
-        Iterator<Resource> resources = request.getResourceResolver().findResources("SELECT * from [pant:module2] AS modules WHERE ISDESCENDANTNODE(\"" +
+        Iterator<Resource> resources = request.getResourceResolver().findResources("SELECT * from [pant:module] AS modules WHERE ISDESCENDANTNODE(\"" +
```
[the remark this answers is not in the document]
Comment (xdavidson) [reply src/main/java/com/redhat/pantheon/servlet/ModuleRootQueryServlet.java]: i think [pant:module2] is a typo. that's why i corrected.  if we are not using it, can we remove it from the code base?

File: src/main/resources/apps/pantheon/modules/all/html.html [hunk 2]
```diff
@@ -42,11 +45,19 @@ <h1>All Modules</h1>
                 <td>
                     <sly data-sly-use.date="${'com.redhat.pantheon.use.DateFormatter' @ format='yyyy/MM/dd HH:mm:ss', date=item.jcr:created}">${date.value}</sly>
                 </td>
+                <td>
+                	
+                </td>
             </tr>
         </sly></sly>
     </table>
+    <div id="bulkOperation"> </div>
```
[the remark this answers is not in the document]
Comment (xdavidson) [reply src/main/resources/apps/pantheon/modules/all/html.html]: yes, it contains a delete button when users selects any checkbox. It stays empty when none of the checkboxes are selected.

File: src/main/resources/apps/pantheon/modules/all/html.html [hunk 3]
```diff
@@ -42,11 +45,19 @@ <h1>All Modules</h1>
...
+    </form>
+</div>
+</div>
+<div data-sly-test="${currentSession.getUserID == 'anonymous'}">
+  Your session expired. Please <a href="/system/sling/login.html">log in</a>.
```
[the remark this answers is not in the document]
Comment (xdavidson) [reply src/main/resources/apps/pantheon/modules/all/html.html]: yes, i ran into that last time I tried to demo to Carlos:-)

File: src/main/resources/apps/pantheon/modules/all/html.html [hunk 4]
```diff
@@ -116,5 +127,38 @@ <h1>All Modules</h1>
...
+	    let btn = document.getElementById("delete");
+		if (! btn) {
+		  deleteButton = document.createElement("input");
+		  deleteButton.setAttribute("type", "submit");
+		  deleteButton.setAttribute("value", "Delete");
+		  deleteButton.setAttribute("id", "delete");
+		  deleteButton.setAttribute("class", "btn btn-primary");
+		  deleteButton.setAttribute("onclick", "form.submit()");
+		  document.getElementById("bulkOperation").appendChild(deleteButton);
+		  form.setAttribute("action", "/content/" + checkbox.id);
```
[the remark this answers is not in the document]
Comment (xdavidson) [reply src/main/resources/apps/pantheon/modules/all/html.html]: yes. I'm using sling API to delete resource(s), not Sling Post Servlet.

File: src/main/java/com/redhat/pantheon/servlet/BulkDeleteServlet.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        		}
+        	}
+        	
+        } 
+        catch (Exception e) {
+        	// Log the error.
+        	logger.error("Module deletion failed: {}", e.getMessage(), e);
+        } finally {
+        	resourceResolver.commit();
+        	response.sendRedirect("/modules.html");
```
[the remark this answers is not in the document]
Comment (xdavidson) [reply src/main/java/com/redhat/pantheon/servlet/BulkDeleteServlet.java]: i added exception handling around .commit(). let me know what you think.

I replaced the hardcoded redirect value with HTTP referrer. The delete operation is handled though sling API. It doesn't route the browser back to the referrer automatically after a successful commit. That's why I added a redirect to take users back to where they were.

File: src/main/java/com/redhat/pantheon/servlet/ModuleRootQueryServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (benradey) [reply src/main/java/com/redhat/pantheon/servlet/ModuleRootQueryServlet.java]: We probably should, but as a separate PR. I just wanted to make sure I wasn't missing anything. For now, don't bother with reverting/removing this class, whatever state it's in, let's leave it there. Thanks!

File: src/main/resources/apps/pantheon/modules/all/html.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (benradey) [reply src/main/resources/apps/pantheon/modules/all/html.html]: Yep I caught that later. Cool!

File: src/main/java/com/redhat/pantheon/servlet/BulkDeleteServlet.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (benradey) [reply src/main/java/com/redhat/pantheon/servlet/BulkDeleteServlet.java]: That wasn't quite what I had in mind so I opened a new PR against your branch, let me know if you agree with that approach.

I really like your referred redirect idea, that's perfect!

File: src/main/resources/apps/pantheon/modules/all/html.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (benradey) [reply src/main/resources/apps/pantheon/modules/all/html.html]: Ah, I just checked out the code and ran a test run myself. I see what's happening. What happens is that the request is made against the last module that you checkboxed - and only that resource - BUT the request includes all of your checkboxes as repeating "module" parameters.

Then in your servlet, you ignore the resource that the request is pointing at. Instead, you loop through the "module" values.

That's an interesting technique. I'm not sure if there's a better way to do it or not, but this works, so, cool!

**Pull request merged**